tools/preflight/preflight.go

```diff
@@ -0,0 +1,620 @@
+
+	co := &CleanupOptions{
+		Ctx:        o.Context,
+		Kubeconfig: o.Kubeconfig,
+		Namespace:  o.Namespace,
+		Logger:     o.Logger,
+	}
+	if !preflightStatus {
+		o.Logger.Warnln("Some preflight checks failed")
+		if o.PerformCleanupOnFail {
+			if err = co.CleanupByUID(resNameSuffix); err != nil {
+				o.Logger.Errorln(fmt.Sprintf("%s Failed to cleanup preflight resources :: %s", cross, err.Error()))
+			}
+		}
+	} else {
+		o.Logger.Infoln("All preflight checks succeeded!")
+		if err = co.CleanupByUID(resNameSuffix); err != nil {
+			o.Logger.Errorln(fmt.Sprintf("%s Failed to cleanup preflight resources :: %s", cross, err.Error()))
+		}
+	}
+}
+
+// checkKubectl checks whether kubectl utility is installed.
+func (o *Options) checkKubectl() error {
+	path, err := goexec.LookPath("kubectl")
+	if err != nil {
+		return fmt.Errorf("error finding 'kubectl' binary in $PATH of the system :: %s", err.Error())
+	}
+	o.Logger.Infoln(fmt.Sprintf("kubectl found at path - %s", path))
+
+	return nil
+}
+
+// checkClusterAccess Checks whether access to kubectl utility is present on the client machine.
+func (o *Options) checkClusterAccess() error {
+	cmdOut, err := shell.RunCmd(fmt.Sprintf("kubectl cluster-info --kubeconfig=%s", o.Kubeconfig))
+	if err != nil {
+		return err
+	} else if cmdOut.ExitCode != 0 {
+		return fmt.Errorf("unable to access kubernetes context :: %s", err.Error())
+	}
+	o.Logger.Infoln(fmt.Sprintf("%s Able to access Kubernetes cluster", check))
+
+	return nil
+}
+
+// checkHelmVersion checks whether minimum helm version is present.
+func (o *Options) checkHelmVersion() error {
+	if isOCPk8sCluster(o.Logger) {
+		o.Logger.Infoln(fmt.Sprintf("%s Running OCP cluster. Helm not needed for OCP clusters", check))
+		return nil
+	}
+	// check whether helm exists
+	path, err := goexec.LookPath("helm")
+	if err != nil {
+		return fmt.Errorf("error finding 'helm' binary in $PATH of the system :: %s", err.Error())
+	}
+	o.Logger.Infoln(fmt.Sprintf("helm found at path - %s", path))
+
+	// check minimum version of elm
+	cmdOut, err := shell.RunCmd("helm version --template '{{.Version}}'")
+	if err != nil {
+		return err
+	}
+	helmVersion := cmdOut.Out[1:]
+	if helmVersion < minHelmVersion {
+		return fmt.Errorf("helm does not meet minimum version requirement.\nUpgrade helm to minimum version - %s", minHelmVersion)
+	}
+	o.Logger.Infoln(fmt.Sprintf("%s Helm version %s meets required version", check, helmVersion))
+
+	return nil
+}
+
+// checkKubernetesVersion checks whether minimum k8s version requirement is met
+func (o *Options) checkKubernetesVersion() error {
+	serverVer, err := clientSet.ServerVersion()
+	if err != nil {
+		return err
+	}
+	major, minor := serverVer.Major, serverVer.Minor
+	rgx := regexp.MustCompile(`[^0-9.]`)
+	minor = rgx.ReplaceAllString(minor, "")
+	minVer := strings.Split(minK8sVersion, ".")
+	minMajor, minMinor := minVer[0], minVer[1]
+	if (major > minMajor) || (major == minMajor && minor >= minMinor) {
+		return nil
+	}
```

Comment:
see if you can use https://github.com/hashicorp/go-version

Reply:
No cannot use that, as minor version received is "19+", so cannot parse this.
Malformed version error is returned by go-version lib